### PR TITLE
fix(secure_storage)!: use correct values for `kSecAttrAccessible`

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
*Issue #, if available:* https://github.com/aws-amplify/amplify-flutter/issues/2535

*Description of changes:*
- Map KeychainAttributeAccessible value to correct values in keychain.
- Gracefully handle the scenario when KeychainAttributeAccessible changes
- Add tests for the updated behavior

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by ragingsquirrel3 — APPROVED** [review]

File: packages/secure_storage/amplify_secure_storage_dart/lib/src/platforms/amplify_secure_storage_cupertino.dart [hunk 1]
```diff
@@ -51,9 +51,9 @@ class AmplifySecureStorageCupertino extends AmplifySecureStorageInterface {
   void write({required String key, required String value}) {
     return using((Arena arena) {
       try {
-        _add(key: key, value: value, arena: arena);
```
Comment (Jordan-Nelson) [inline packages/secure_storage/amplify_secure_storage_dart/lib/src/platforms/amplify_secure_storage_cupertino.dart:54]: Note: This change isn't related to the bug fix and has no real impact on behavior.

While reviewing Amplify-iOS code, I noticed they first attempt to update the key, and then add it if the key doesn't exist. We were doing the opposite. The update aligns closer with Amplify-swift and is slightly more efficient for updates, which we would expect to occur more frequently than insertions.

File: packages/secure_storage/amplify_secure_storage/example/integration_test/cupertino_test.dart [hunk 2]
```diff
@@ -74,4 +76,92 @@ void main() {
...
+        storage1.removeAll();
+        storage2.removeAll();
+      });
+      test(
+        'read returns null when reading keys with a different kSecAttrAccessible value',
+        () async {
+          storage1.write(key: key1, value: value1);
+
+          expect(storage1.read(key: key1), value1);
+          expect(storage2.read(key: key2), isNull);
```
Comment (dnys1) [inline packages/secure_storage/amplify_secure_storage/example/integration_test/cupertino_test.dart:121]: Should this be `key1`?

File: packages/secure_storage/amplify_secure_storage/example/integration_test/cupertino_test.dart [hunk 3]
```diff
@@ -74,4 +76,92 @@ void main() {
...
+          storage2.delete(key: key1);
+
+          expect(storage1.read(key: key1), isNull);
+        },
+      );
+
+      test(
+        'removeAll can delete keys regardless of the kSecAttrAccessible value',
+        () async {
+          storage1.write(key: key1, value: value1);
```
Comment (dnys1) [inline packages/secure_storage/amplify_secure_storage/example/integration_test/cupertino_test.dart:141]: Should the reverse also be tested? `storage2 write / storage1 removeAll`

File: packages/secure_storage/amplify_secure_storage_dart/lib/src/interfaces/amplify_secure_storage_interface.dart [hunk 4]
```diff
@@ -21,7 +21,7 @@ abstract class AmplifySecureStorageInterface extends SecureStorageInterface
   /// {@template amplify_secure_storage_dart.amplify_secure_storage_interface.remove_all}
   /// Removes all key-value pairs for the current scope.
   ///
-  /// Only available on Linux. Will throw [UnimplementedError]
+  /// Only available on iOS, MacOs, and Linux. Will throw [UnimplementedError]
```
Comment (dnys1) [inline packages/secure_storage/amplify_secure_storage_dart/lib/src/interfaces/amplify_secure_storage_interface.dart:24]:
```suggestion
  /// Only available on iOS, macOS, and Linux. Will throw [UnimplementedError]
```

File: packages/secure_storage/amplify_secure_storage/example/integration_test/cupertino_test.dart [hunk 5]
```diff
@@ -16,18 +17,19 @@ const key2 = 'key_2';
...
-
-/// MacOS & iOS app uninstall & re-install tests.
-///
-/// These tests are only relevant for amplify_secure_storage
-/// (not amplify_secure_storage_dart) as they use
-/// NSUserDefaultsAPI via pigeon.
 void main() {
   IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  /// MacOS & iOS app uninstall & re-install tests.
```
Comment (dnys1) [inline packages/secure_storage/amplify_secure_storage/example/integration_test/cupertino_test.dart:23]:
```suggestion
  /// macOS & iOS app uninstall & re-install tests.
```

File: packages/secure_storage/amplify_secure_storage/example/integration_test/cupertino_test.dart [hunk 6]
```diff
@@ -16,18 +17,19 @@ const key2 = 'key_2';
...
+  ///
+  /// These tests are only relevant for amplify_secure_storage
+  /// (not amplify_secure_storage_dart) as they use
+  /// NSUserDefaultsAPI via pigeon.
```
Comment (dnys1) [inline packages/secure_storage/amplify_secure_storage/example/integration_test/cupertino_test.dart:27]: The new tests can be done in Dart too, though, no?

**Review by dnys1 — APPROVED** [review]

File: packages/secure_storage/amplify_secure_storage/example/integration_test/cupertino_test.dart (hunk 2 above)
Comment (Jordan-Nelson) [reply packages/secure_storage/amplify_secure_storage/example/integration_test/cupertino_test.dart]: Oops, yes. Updated. Thanks.

File: packages/secure_storage/amplify_secure_storage/example/integration_test/cupertino_test.dart (hunk 6 above)
Comment (Jordan-Nelson) [reply packages/secure_storage/amplify_secure_storage/example/integration_test/cupertino_test.dart:27]: They probably could with some extra setup. On macOS they require keychain entitlement, which makes them easier to run from a flutter app.

**Pull request merged**